kotsadm/web/src/components/apps/AppGitops.jsx

```diff
@@ -105,33 +102,22 @@ class AppGitops extends Component {
           Utilities.logoutUser();
           return;
         }
-        let msg = `Unexpected status code: ${res.status}`;
+        this.props.refetch();
+
         if (res.status === 400) {
-          msg = `Unable to authenticate. Please make sure you have added your deployment key to the git repo.`;
-        }
-        try {
-          const data = await res.json();
-          if (data?.error) {
-            console.log(`Failed to test gitops connection: ${data.error}`);
+          const response = await res.json();
+          if (response?.error) {
+            console.log(response?.error);
           }
-        } catch (err) {
-          console.log(`Failed to test gitops connection: ${err}`);
+          throw new Error(`authentication failed`);
         }
-        this.setState({
-          errorTitle: "Failed to test connection",
-          errorMsg: msg,
-          displayErrorModal: true,
-        });
-        this.props.refetch();
-        return;
+        throw new Error(`unexpected status code: ${res.status}`);
```

Comment:
i think we should keep handling the error here instead of throwing it because this function is also called when clicking `Try again`

Reply:
I do handle it in the outer try catch